File: diretorioprincipal/questao2.java
//atividade 06/09/24
//aluno: João Pedro Araújo Inácio
//questao2

public class questao2 {

    static class Carro {
        String marca;
        String modelo;
        int ano;
        int velocidade;

        public Carro(String marca, String modelo, int ano) {
            this.marca = marca;
            this.modelo = modelo;
            this.ano = ano;
            this.velocidade = 0; 
        }

        public void acelerar(int incremento) {
            this.velocidade += incremento;
            System.out.println(this.modelo + " acelerou para " + this.velocidade + " km/h");
        }

        public void frear(int decremento) {
            this.velocidade -= decremento;
            if (this.velocidade < 0) {
                this.velocidade = 0;
            }
            System.out.println(this.modelo + " desacelerou para " + this.velocidade + " km/h");
        }

        public void exibirVelocidade() {
            System.out.println("Velocidade atual do " + this.modelo + ": " + this.velocidade + " km/h");
        }
    }

    public static void main(String[] args) {

        Carro carro1 = new Carro("volkswagen", "polo", 2009);

        carro1.acelerar(30);
        carro1.frear(10);
        carro1.exibirVelocidade();
    }
}

File: diretorioprincipal/questao2.py
#atividade 06/09/24
#aluno: João Pedro Araújo Inácio
#questao2

class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0 

    def acelerar(self, incremento):
        self.velocidade += incremento
        print(f"{self.modelo} acelerou para {self.velocidade} km/h")

    def frear(self, decremento):
        self.velocidade -= decremento
        if self.velocidade < 0:
            self.velocidade = 0
        print(f"{self.modelo} desacelerou para {self.velocidade} km/h")

    def exibir_velocidade(self):
        print(f"Velocidade atual do {self.modelo}: {self.velocidade} km/h")


carro1 = Carro("volkswagen", "Polo", 2009)

carro1.acelerar(30)
carro1.frear(10)
carro1.exibir_velocidade()
